---
title: "A Biophysics-based Protein Language Model for Protein Engineering"
date: 2024-06-18
author: "Dr. Sam Gelman"
image: "https://img.youtube.com/vi/milM6-2RF18/maxresdefault.jpg"
description: "We introduce Mutational Effect Transfer Learning (METL), a specialized protein language model that bridges the gap between traditional biophysics-based and machine learning approaches by incorporating synthetic data from molecular simulations."

language: 
  title-block-author-plural: "Presenters"
  title-block-published: "Date"
categories:
  - Cross Labs AI
  - Transfer learning
  - Biophysics
  - Protein language models
  - LLM
  - Deep learning
  - Protein engineering
---
Video description from Cross Labs AI:

> Just as words combine to form sentences that convey meaning in human languages, the specific arrangement of amino acids in proteins can be viewed as an information-rich language describing molecular structure and behavior. 
> 
> Protein language models harness advances in natural language processing to decode intricate patterns and relationships within protein sequences. These models learn meaningful, low-dimensional representations that capture the semantic organization of protein space and have broad utility in protein engineering. However, while protein language models are powerful, they do not take advantage of the extensive knowledge of protein biophysics and molecular mechanisms acquired over the last century. Thus, they are largely unaware of the underlying physical principles governing protein function.
> 
> We introduce Mutational Effect Transfer Learning (METL), a specialized protein language model that bridges the gap between traditional biophysics-based and machine learning approaches by incorporating synthetic data from molecular simulations. We pretrain a transformer on millions of molecular simulations to capture the relationship between protein sequence, structure, energetics, and stability. We then finetune the neural network to harness these fundamental biophysical signals and apply them when predicting protein functional scores from experimental assays. METL excels in protein engineering tasks like generalizing from small training sets and extrapolating to new sequence positions. We demonstrate METL's ability to design functional green fluorescent protein variants when trained on only 64 experimental examples.

### Links
- About the Speaker â†’ [samgelman.com](https://samgelman.com)
- Check out the [preprint](https://www.biorxiv.org/content/10.1101/2024.03.15.585128v1) a
#### Software
A collection of METL software repositories is provided to reproduce the results of this manuscript and run METL on new data:

- https://github.com/gitter-lab/metl for pretraining and finetuning METL PLMs (archived at doi:10.5281/zenodo.10819483)

- https://github.com/gitter-lab/metl-sim for generating biophysical attributes with Rosetta (archived at doi:10.5281/zenodo.10819523)

- https://github.com/gitter-lab/metl-pretrained for making predictions with pretrained METL PLMs (archived at doi:10.5281/zenodo.10819499)

- https://github.com/gitter-lab/metl-pub for additional code and data to reproduce these results (archived at doi:10.5281/zenodo.10819536)

All code is available under the MIT license.

  

{{< video https://www.youtube.com/watch?v=milM6-2RF18 >}}

#### Jump to section
- [3:01, Intro](https://www.youtube.com/watch?v=milM6-2RF18&t=181s)
- [5:03, Proteins as nature's molecular machines](https://www.youtube.com/watch?v=milM6-2RF18&t=307s)
- [6:00, create or modify functions to fulfill new functions](https://www.youtube.com/watch?v=milM6-2RF18&t=360s)
- [6:58, Proteins defined by a sequence of amino acides](https://www.youtube.com/watch?v=milM6-2RF18&t=360s)
- [10:34 Challenge: Vastness of sequence space](https://www.youtube.com/watch?v=milM6-2RF18&t=634s)
- [11:18: Navigating sequence space](https://www.youtube.com/watch?v=milM6-2RF18&t=678s)
- [11:52, Challenge: Small changes can have a large impact](https://www.youtube.com/watch?v=milM6-2RF18&t=712s)
- [12:51, Protein language models (PLMs)](https://www.youtube.com/watch?v=milM6-2RF18&t=771s)
- [15:57, A different approach to PLMs: incorporating biophysics](https://www.youtube.com/watch?v=milM6-2RF18&t=957s)
- [17:33, Mutational Effect Transfer Learning (METL)](https://www.youtube.com/watch?v=milM6-2RF18&t=1053s)
- [19:50, Simulating protein structures with Rosetta](https://www.youtube.com/watch?v=milM6-2RF18&t=1190s)
- [21:10, Local and global strategies for simulations](https://www.youtube.com/watch?v=milM6-2RF18&t=1270s)
  - [21:48, Local simulated data](https://www.youtube.com/watch?v=milM6-2RF18&t=1308s)
  - [23:35, Global simulated data](https://www.youtube.com/watch?v=milM6-2RF18&t=1415s)
- [24:26, Train transformer encoder to predict Rosetta scores ](https://www.youtube.com/watch?v=milM6-2RF18&t=1466s)
- [25:38, Finetune to predict experimental fitness score](https://www.youtube.com/watch?v=milM6-2RF18&t=1538s)
- [27:10, Evaluation baselines (evolutionary models): METL, ESM, and EVE](https://www.youtube.com/watch?v=milM6-2RF18&t=1630s)
- [28:26, Generalizing from small datasets](https://www.youtube.com/watch?v=milM6-2RF18&t=1706s)
- [31:08, Extrapolating bond train set](https://www.youtube.com/watch?v=milM6-2RF18&t=1868s)
- [33:50, Simulating specific functions](https://www.youtube.com/watch?v=milM6-2RF18&t=2030s)
- [35:46, How much simulated/experimental data is needed?](https://www.youtube.com/watch?v=milM6-2RF18&t=2146s)
- [38:01, Engineering GFP variants with METL](https://www.youtube.com/watch?v=milM6-2RF18&t=2281s)
- [41:40, Q&A](https://www.youtube.com/watch?v=milM6-2RF18&t=2500s)

